tests/unit/s2n_connection_size_test.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "tls/s2n_connection.h"
+
+bool is_32_bit_platform()
+{
+    return (sizeof(void *) == 4);
+}
+
+/* Test s2n_connection does not grow too much.
+ * s2n_connection is a very large structure. We should be working to reduce its
+ * size, not increasing it.
+ * This test documents changes to its size for reviewers so that we can
+ * make very deliberate choices about increasing memory usage.
+ *
+ * We can't easily enforce an exact size for s2n_connection because it varies
+ * based on some settings (like how many KEM groups are supported).
+ */
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* We don't run this test on 32 bit platforms. The goal of this test is
+     * accomplished as long as it is running on any platform in our CI, and
+     * just running it on a single platform keeps us from having to maintain
+     * multiple static constants.
+    */
+    if (is_32_bit_platform()) {
+        return 0;
+    }
```

Comment:
You should probably still call END_TEST(). It doesn't really matter anymore with cmake swallowing the output, but consistency.